notebooks/Notebook2.md

```diff
@@ -2,33 +2,56 @@
+
+Next I started to simplify the player type classes with individual setters per input type vs one big setter that sets all input types. This was fairly straightforward, I got rid of alot of bulk in the abstract player class and allowed for each player type to be created with just the input of the players number instead of name and number and all the other components that each player has. Then I built setters for each player's unique inputs. This allowed me to write something similar to the following:
+New PassingPlayer(playerOrNum).setYards(numYards).setINCs(outcome).setTDs(outcome).setINTs(outcome).updatePlayer();
+This was useful because I was able to handle inputs that are the same across different methods. ([code here](https://github.com/hmc-cs111-spring2023/BBeckerArtifact/tree/main/src/Player))
+
+Finally I built a parser that takes each part of the input language and utilizes the Game class add function to add a player into the data structure. This involved building code to determine what type of playtype each input from the user was. These functions are my is…() functions. They return a boolean value if the input matches a possible input string. This could be further simplified by using another type of parser but for now It works great. Next I worked on function that are designed to add the player to the game and return the game to the parser these function utilize the add function from the game class to add a new player to the data structure or if the player already exists update the stats by merging the players stats that were just created with the old ones. The individualized settlers for each player type came in really nice for this aspect. Finally I worked on some functions that would transform strings to numbers and parsing out yardage from a string. For example, if the user inputs 22y representing 22 yards the parser needs to transform that into just an integer number 22. ([code here](https://github.com/hmc-cs111-spring2023/BBeckerArtifact/blob/main/src/Parser/Parser.java))
+
 
 ## Questions
 
 **What is the most pressing issue for your project? What design decision do
 you need to make, what implementation issue are you trying to solve, or how
 are you evaluating your design and implementation?**
 
+The most pressing issue right now is to get my repo set up in a way that allows for me to build a physical interface for the user to input the language. I think this is the best way for the user to use my dsl and it's better than using a command line input. Also I will begin to work on error handling for the language next week and I think that this will best be presented to the user with such an interface. 
```

Comment:
Great job addressing my concern about error handling from above as something you'll be doing this upcoming week! I know you've considered the users' backgrounds in your implementation so it's always good to keep those in mind but I think your parser's current implementation with the groupings should make it easier to identify which, parts of the input were faulty and thus you'll be able to give users clearer explanation on what specific part of their input was incorrect.